lib/Runtime/Library/JavascriptBigInt.cpp

```diff
@@ -113,26 +156,182 @@ namespace Js
+    }
+
+    JavascriptBigInt * JavascriptBigInt::AbsoluteIncrement(JavascriptBigInt * pbi)
+    {
+        JavascriptBigInt* result = pbi;
+        digit_t carry = 1;
+        for (digit_t i = 0; i < result->m_length && carry > 0; i++)
+        {
+            digit_t tempCarry = 0;
+            result->m_digits[i] = JavascriptBigInt::AddDigit(result->m_digits[i], carry, &tempCarry);
+            carry = tempCarry;
+        }
+        if (carry > 0) //increase length
+        {
+            if (result->m_length >= result->m_maxLength && !result->Resize(result->m_length + 1))
+            {
+                AssertOrFailFastMsg(false, "AbsoluteIncrement overflow");
+            }
+            result->m_digits[result->m_length++] = carry;
+        }
+        return result;
+    }
+
+    JavascriptBigInt * JavascriptBigInt::AbsoluteDecrement(JavascriptBigInt * pbi)
+    {
+        JavascriptBigInt* result = pbi;
+        Assert(!JavascriptBigInt::IsZero(result));
+        digit_t borrow = 1;
+        for (digit_t i = 0; i < result->m_length && borrow > 0; i++)
+        {
+            digit_t tempBorrow = 0;
+            result->m_digits[i] = JavascriptBigInt::SubDigit(result->m_digits[i], borrow, &tempBorrow);
+            borrow = tempBorrow;
+        }
+        Assert(borrow == 0);
+        // remove trailing zero
+        if (result->m_digits[result->m_length-1] == 0)
+        {
+            result->m_length--;
+        }
+        return result;
+    }
+
+    JavascriptBigInt * JavascriptBigInt::Increment(JavascriptBigInt * pbi)
+    {
+        if (pbi->m_isNegative)
+        {
+            // return 0n for -1n
+            if (pbi->m_length == 1 && pbi->m_digits[0] == 1)
+            {
+                JavascriptBigInt* result = pbi;
+                result->m_digits[0] = 0;
+                result->m_isNegative = false;
+                return result;
+            }
+            return JavascriptBigInt::AbsoluteDecrement(pbi);
+        }
+        return JavascriptBigInt::AbsoluteIncrement(pbi);
+    }
+
+    JavascriptBigInt * JavascriptBigInt::Decrement(JavascriptBigInt * pbi)
+    {
+        if (pbi->m_isNegative)
+        {
+            return JavascriptBigInt::AbsoluteIncrement(pbi);
+        }
+        if (JavascriptBigInt::IsZero(pbi)) // return -1n for 0n
+        {
+            JavascriptBigInt* result = pbi;
+            result->m_digits[0] = 1;
+            result->m_isNegative = true;
+            return result;
+        }
+        return JavascriptBigInt::AbsoluteDecrement(pbi);
+    }
+
+    Var JavascriptBigInt::Increment(Var aRight)
+    {
+        AssertMsg(VarIs<JavascriptBigInt>(aRight), "BigInt Increment");
+
+        JavascriptBigInt* rightBigInt = VarTo<JavascriptBigInt>(aRight);
+        return JavascriptBigInt::Increment(rightBigInt);
+    }
+
+    Var JavascriptBigInt::Decrement(Var aRight)
+    {
+        AssertMsg(VarIs<JavascriptBigInt>(aRight), "BigInt Increment");
```

Comment:
Decrement

Reply:
Also, VarTo already asserts this, so feel free to remove if you like.

---
In reply to: [226474297](https://github.com/Microsoft/ChakraCore/pull/5788#discussion_r226474297) [](ancestors = 226474297)